manta-accounting/src/wallet/signer/nullifier_map.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Nullifier Map
+
+use alloc::{collections::BTreeSet, vec::Vec};
+
+#[cfg(feature = "std")]
+use {core::hash::Hash, std::collections::HashSet};
+
+/// Nullifier Map
+pub trait NullifierMap<T>: Default {
+    /// Creates a new [`NullifierMap`].
+    fn new() -> Self;
+
+    /// Inserts `item` in `self`.
+    fn insert(&mut self, item: T) -> bool;
```

Comment:
i'd also add len() function in case we need to debug in the future, like in `wasm_debugging_utilities`

Reply:
good idea